File: bigBangClient/Assets/Babu/Editor/PackageBuild/Base/BuildConfig.cs
using LightJson;
using System;
using UnityEditor;
using UnityEngine;

namespace Babu.Editor.Build
{
    public class BuildConfig
    {
        public JsonValue Config;

        public void Load(string filePath, string channelId)
        {
            TextAsset textAsset = AssetDatabase.LoadAssetAtPath<TextAsset>(filePath);
            if (textAsset == null)
            {
                throw new Exception($"Can Not Find: {filePath}");
            }

            JsonValue root = JsonValue.Parse(textAsset.text);
            Config = root;
            if (root.AsJsonObject.ContainsKey("default"))
            {
                // 如果有default，优先使用default
                Config = root["default"];
            }
            if (root.AsJsonObject.ContainsKey(channelId))
            {
                // 合并channel的特殊配置
                Merge(Config, root[channelId]);
            }

            Debug.Log($"{filePath} Config Load Succ: " + Config.ToString(true));
        }

        void Merge(JsonValue config, JsonValue channelConfig)
        {
            foreach (var iter in channelConfig.AsJsonObject)
            {
                if (iter.Value.IsJsonObject)
                {
                    Merge(config[iter.Key], iter.Value);
                }
                else
                {
                    config[iter.Key] = iter.Value;
                }
            }
        }

        public void Load(string filePath)
        {
            Load(filePath, BuildArgs.Instance.ChannelId);
        }

        public string GetConfig(string configName)
        {
            return Config[configName].AsString;
        }
        public string GetConfig(string configName1, string configName2)
        {
            return Config[configName1][configName2].AsString;
        }
    }
}

File: bigBangClient/Assets/Babu/BuildScript/BuildConfig.py
import json
from Utils import log

class BuildConfig:
    def __init__(self, build_args):
        self.build_args = build_args
    
    def init(self):
        with open("Assets/Babu/BuildConfig/BabuBuildConfig.json", 'r', encoding="utf-8") as f:
            if not f:
                return False
            self.config = root = json.load(f)
            if "default" in root:
                self.config = root["default"]
            if self.build_args.channel_id in root:
                self.merge(self.config, root[self.build_args.channel_id])
            log('==========build config init succ: ' + json.dumps(self.config, default=lambda obj: obj.__dict__, sort_keys=True))
            return True

    def merge(self, config, chanel_config):
        for key in chanel_config.keys():
            if type(chanel_config[key]).__name__ == 'dict':
                self.merge(config[key], chanel_config[key])
            else:
                config[key] = chanel_config[key]
    
    def getConfig(self, config_name):
        attrs = config_name.split(".")
        config = self.config
        for i in range(len(attrs)):
            config = config[attrs[i]]
        return config